testing/upload_validation.ts

```diff
@@ -355,1385 +365,1409 @@ const setEndpointPostSetupValidator = zodCompleteObject<SetEndpoint>({
   'Either getOptions or getOptionsFormula must be specified.',
 );
 
-const singleAuthDomainSchema = z
-  .string()
-  .nonempty()
-  .refine(domain => domain.indexOf(' ') < 0, {
-    message: 'The `networkDomain` in setUserAuthentication() cannot contain spaces. Use an array for multiple domains.',
-  });
-
-const baseAuthenticationValidators = {
-  // TODO(jonathan): Remove these after fixing/exporting types for Authentication metadata, as they're only present
-  // in the full bundle, not the metadata.
-  getConnectionName: z.unknown().optional(),
-  getConnectionUserId: z.unknown().optional(),
-  instructionsUrl: z.string().optional(),
-  requiresEndpointUrl: z.boolean().optional(),
-  endpointDomain: z.string().optional(),
-  // The items are technically a discriminated union type but that union currently only has one member.
-  postSetup: z.array(setEndpointPostSetupValidator).optional(),
-  networkDomain: z.union([singleAuthDomainSchema, z.array(singleAuthDomainSchema).nonempty()]).optional(),
-};
+interface buildMetadataSchemaArgs {
+  sdkVersion?: string;
```

Comment:
We don't always have an sdkVersion readily available in the calling context, so I kept it optional for simplicity. You can see in the other PR where I used `{}` instead of `{sdkVersion: ...}`